src/core/qgsexpressioncontext.cpp

```diff
@@ -630,21 +630,12 @@ QgsExpressionContextScope* QgsExpressionContextUtils::projectScope()
 
   QgsExpressionContextScope* scope = new QgsExpressionContextScope( QObject::tr( "Project" ) );
 
-  //add variables defined in project file
-  QStringList variableNames = project->readListEntry( "Variables", "/variableNames" );
-  QStringList variableValues = project->readListEntry( "Variables", "/variableValues" );
+  QgsStringMap vars = QgsProject::instance()->variables();
 
-  int varIndex = 0;
-  Q_FOREACH ( const QString& variableName, variableNames )
```

Comment:
Just curious,  why you dropped Q_FOREACH in favour of the iterator?
Is the iterator approach in general recommended over Q_FOREACH?